kroma-chain-ops/cmd/check-kroma-mpt/main.go

```diff
@@ -0,0 +1,627 @@
+	}
+
+	env.log.Info("L1Block address and tx data test: SUCCESS")
+	return nil
+}
+
+func checkL1Block(ctx context.Context, env *actionEnv) error {
+	cl, err := bindings.NewL1Block(oppredeploys.L1BlockAddr, env.l2)
+	if err != nil {
+		return fmt.Errorf("failed to create bindings around L1Block contract: %w", err)
+	}
+
+	blobBaseFee, err := cl.BlobBaseFee(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get blob basfee from L1Block contract: %w", err)
+	}
+	if big.NewInt(0).Cmp(blobBaseFee) == 0 {
+		return errors.New("blob basefee must never be 0, EIP specifies minimum of 1")
+	}
+
+	rollupCfg, err := env.rollupCl.RollupConfig(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve rollup config: %w", err)
+	}
+	l2RPC := client.NewBaseRPCClient(env.l2.Client())
+	l2EthCl, err := sources.NewL2Client(l2RPC, env.log, nil, sources.L2ClientDefaultConfig(rollupCfg, false))
+	if err != nil {
+		return fmt.Errorf("failed to create eth client: %w", err)
+	}
+
+	prevValRewardScalar, err := l2EthCl.GetStorageAt(ctx, oppredeploys.L1BlockAddr, types.KromaL1BlobBaseFeeSlot, "latest")
+	if err != nil {
+		return fmt.Errorf("failed to get previous validatorRewardScalar storage: %w", err)
+	}
+	if big.NewInt(0).Cmp(prevValRewardScalar.Big()) != 0 {
+		return errors.New("validatorRewardScalar must be 0")
+	}
+
+	env.log.Info("L1Block contract test: SUCCESS")
+	return nil
+}
+
+func checkGPO(ctx context.Context, env *actionEnv) error {
+	cl, err := bindings.NewGasPriceOracle(predeploys.GasPriceOracleAddr, env.l2)
+	if err != nil {
+		return fmt.Errorf("failed to create bindings around GasPriceOracle contract: %w", err)
+	}
+
+	isKromaMPT, err := cl.IsKromaMPT(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get Kroma MPT status: %w", err)
+	}
+	if !isKromaMPT {
+		return fmt.Errorf("GPO is not set to Kroma MPT: %w", err)
+	}
+
+	l1BaseFee, err := cl.L1BaseFee(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get l1 base fee: %w", err)
+	}
+	if l1BaseFee.Cmp(common.Big0) == 0 {
+		return errors.New("l1 base fee should not be zero")
+	}
+
+	blobBaseFee, err := cl.BlobBaseFee(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get blob base fee: %w", err)
+	}
+	if blobBaseFee.Cmp(common.Big0) == 0 {
+		return errors.New("blob base fee should not be zero")
+	}
+
+	baseFeeScalar, err := cl.BaseFeeScalar(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get base fee scalar: %w", err)
+	}
+	if baseFeeScalar == 0 {
+		return errors.New("base fee scalar should not be zero")
+	}
+
+	blobBaseFeeScalar, err := cl.BlobBaseFeeScalar(nil)
+	if err != nil {
+		return fmt.Errorf("failed to get blob base fee scalar: %w", err)
+	}
+	if blobBaseFeeScalar == 0 {
+		return errors.New("blob base fee scalar should not be zero")
+	}
+
+	env.log.Info("GasPriceOracle contract test: SUCCESS")
+	return nil
+}
+
+// assuming a 0 (fail) or non-zero (success) on the stack, this performs a revert or self-destruct
+func conditionalCode(data []byte) []byte {
```

Comment:
I guess based on the comment, it can execute `selfdestruct` when the call succeeds, so how about just uncomment the L#434 and perform an empty call?